airflow/api_fastapi/core_api/routes/ui/grid.py

```diff
@@ -0,0 +1,324 @@
+        )
+
+    current_time = timezone.utcnow()
+    # Retrieve, sort and encode the previous DAG Runs
+    base_query = (
+        select(
+            DagRun.run_id,
+            DagRun.queued_at,
+            DagRun.start_date,
+            DagRun.end_date,
+            DagRun.state,
+            DagRun.run_type,
+            DagRun.data_interval_start,
+            DagRun.data_interval_end,
+            DagRun.dag_version_id.label("version_number"),
+        )
+        .select_from(DagRun)
+        .where(DagRun.dag_id == dag.dag_id, DagRun.logical_date <= func.coalesce(base_date, current_time))
+        .order_by(DagRun.id.desc())
+    )
+
+    def get_dag_run_sort_param():
+        """Get the Sort Param for the DAG Run."""
+
+        def _get_run_ordering_expr(name: str) -> ColumnOperators:
+            """Get the Run Ordering Expression."""
+            expr = DagRun.__mapper__.columns[name]
+            # Data interval columns are NULL for runs created before 2.3, but SQL's
+            # NULL-sorting logic would make those old runs always appear first. In a
+            # perfect world we'd want to sort by ``get_run_data_interval()``, but that's
+            # not efficient, so instead the columns are coalesced into logical_date,
+            # which is good enough in most cases.
+            if name in ("data_interval_start", "data_interval_end"):
+                expr = func.coalesce(expr, DagRun.logical_date)
+            return expr.desc()
+
+        ordering_expression = (_get_run_ordering_expr(name) for name in dag.timetable.run_ordering)
+        # create SortParam with ordering_expression and DagRun.id.desc()
+        return ordering_expression
+
+    dag_runs_select_filter, _ = paginated_select(
+        statement=base_query.order_by(*get_dag_run_sort_param(), DagRun.id.desc()),
+        filters=[
+            run_types,
+            run_states,
+        ],
+        order_by=None,
+        offset=offset,
+        limit=num_runs,
+    )
+
+    dag_runs = session.execute(dag_runs_select_filter)
+
+    # Check if there are any DAG Runs with given criteria to eliminate unnecessary queries/errors
+    if not dag_runs:
+        return GridResponse(dag_runs=[])
+
+    # Retrieve, sort and encode the Task Instances
+    tis_of_dag_runs, _ = paginated_select(
+        statement=select(
+            TaskInstance.run_id,
+            TaskInstance.task_id,
+            TaskInstance.try_number,
+            TaskInstance.state,
+            TaskInstance.start_date,
+            TaskInstance.end_date,
+            TaskInstance.queued_dttm.label("queued_dttm"),
+        )
+        .join(TaskInstance.task_instance_note, isouter=True)
+        .where(TaskInstance.dag_id == dag.dag_id),
+        filters=[],
+        order_by=SortParam(allowed_attrs=["task_id", "run_id"], model=TaskInstance).dynamic_depends(
+            "task_id"
+        )(),
```

Comment:
`dynamic_depends` is only useful for FastAPI dependency injection system. When we manuallay want an instance of if you can just:

```python
SortParam(allowed_attrs=["task_id", "run_id"], model=TaskInstance).set_value('your_value').
```

Reply:
Updated like this